keras_cv/layers/preprocessing/channel_shuffle.py

```diff
@@ -0,0 +1,103 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf 
+from tensorflow.keras import layers, backend
+
+class ChannelShuffle(layers.Layer):
+    """ChannelShuffle performs shuffling the channel of inputs.  
+
+    Input shape:
+        The expected images should be [0-255] pixel ranges.
+        3D (unbatched) or 4D (batched) tensor with shape:
+        `(..., height, width, channels)`, in `"channels_last"` format
+    Output shape:
+        3D (unbatched) or 4D (batched) tensor with shape:
+        `(..., height, width, channels)`, in `"channels_last"` format
+
+    Args:
+        groups:
+            Number of groups to divide the input channels. Default 3. 
+        seed:
+            Integer. Used to create a random seed.
+
+    Usage:
+    ```python
+    (images, labels), _ = tf.keras.datasets.cifar10.load_data()
+    channelshuffle = keras_cv.layers.preprocessing.ChannelShuffle()
+    augmented_images = channelshuffle(images)
+    ```
+    """
+    def __init__(
+        self,
+        groups=3,
+        seed=None,
+        **kwargs
+    ):
+        super().__init__(**kwargs)
+        self.groups=groups
+        self.seed = seed
+
+    def _channel_shuffling(self, images):
+        unbatched = images.shape.rank == 3
```

Comment:
we can't use `.shape` in graph mode.  I believe we need to use `tf.shape`.

Reply:
I think it's safe. Check here https://github.com/keras-team/keras/blob/v2.8.0/keras/layers/preprocessing/image_preprocessing.py#L861-L872

Reply:
Right, `shape.rank` will work, however e.g. `shape[3]` might be None.